### ISSUE TITLE
BambuStudio adds holes in the model. 

### ISSUE THREAD

**Issue by Babsobar** [bambulab/BambuStudio#1442]
**Describe the bug**
In prints that have thin walls, Bambustudio adds holes in the design. Turning "detect thin walls" on adds holes in other places.

**3mf File for This Bug**
Github does not allow the sharing of .3MF files. I will share the whole of the project in a dropbox or whatever.
https://file.io/cGhxBk59vLJh

**To Reproduce**
In fusion 360, Make a file in which the infill is part of the design of the piece, meaning that you have manually designed the infill as part of the piece needed to be printed. Shell the piece so that the outside walls are 0.45 mm. 
The use case for this is for 3d printed parts that are transparent. See photos. 
Set filament to PLA, and 0,16mm OPTIMAL profile. Set infill to 0%. Slice. Send to printer over the internet, using a new roll of natural PLA from fillamentum.

**Expected behavior**
Have bambustudio keep the file as it is meant to be! This does not happen in prusaSlicer. Also, the print quality of the surface is absolutely terrible! It produces "scratches" on the surface that are protruding almost 0.5 mm. See photo. 

**Screenshots**
![Defect](https://user-images.githubusercontent.com/17145038/224059375-c96909af-adef-4344-84de-69efb72868c0.PNG)
![Defect2](https://user-images.githubusercontent.com/17145038/224059405-a05997d8-bc69-4de9-bd82-978aa208d09f.PNG)
When detect thin walls is on:
![Detectthinwalls](https://user-images.githubusercontent.com/17145038/224059518-7f138350-e267-4fef-a9b4-d75f5f107441.PNG)
![Holes in print](https://user-images.githubusercontent.com/17145038/224059552-0ef839f4-c38e-46df-acd8-d2067b9004fa.PNG)
![PXL_20230309_124001376](https://user-images.githubusercontent.com/17145038/224059606-43b0fb91-0acf-4aa5-8ef0-ffe71670b8c1.jpg)
Terrible print surface quality 
![PXL_20230309_120833641](https://user-images.githubusercontent.com/17145038/224059725-415483e1-7f06-4436-905c-95c5a5fa2a13.jpg)
![PXL_20230309_124001376](https://user-images.githubusercontent.com/17145038/224059794-4e24d126-54f9-4073-915d-233a0b73ea63.jpg)

**Desktop (please complete the following information):**
 - Windows 10, latest updates
 - Latest Bambulab version

I bought this machine as a professional artist, because I wanted to have a more detailed, faster, and modern machine, but compared to when I had a Prusa MK2S and I was using Prusaslicer, I have been having more problems. I know that what I print is not the usual but this machine is supposed to be a professional tool and right now, it really is putting a strain on my professional activity because of strange problems like this that I didn't have before.

**Comment by SaltWei** [comment]
> Github does not allow the sharing of .3MF files. I will share the whole of the project in a dropbox or whatever.
> https://file.io/cGhxBk59vLJh

Can not get the file. You can just rename the 3MF as .zip

**Comment by Babsobar** [comment]
Thank you for your answer, here is the file zipped up.

[Troubleshooting.zip](https://github.com/bambulab/BambuStudio/files/10972201/Troubleshooting.zip)

**Comment by SaltWei** [comment]
![image](https://user-images.githubusercontent.com/110660407/225186156-d2dc06ac-075f-4929-b8fb-e7b8e4bb45e7.png)
![image](https://user-images.githubusercontent.com/110660407/225186197-0e2c53ab-fc26-4c12-9ecc-0f49bfbe5318.png)
![image](https://user-images.githubusercontent.com/110660407/225186227-ab4d25a9-01a3-4d50-b846-bef888580d16.png)
It is just caused by too thin model. Not slicer bug.
Changing settings as pictures can help.

**Comment by Babsobar** [comment]
![prusa](https://user-images.githubusercontent.com/17145038/225553939-bfa50c7a-6fdf-4774-88f5-303ca016ad4d.PNG)
This is the same file in Prusaslicer, as you can see, no holes are generated. 

![Same problems in bambustudio](https://user-images.githubusercontent.com/17145038/225554043-d3d8dccd-d245-4ba5-91fc-bf4af6b43d27.PNG)
I have tried your setting in Bambustudio, the only one that helps is Arachne, but I need the printer to slow down for overhangs because I can't use supports for this print. Seeing as this is a transparent print and has 

"It is just caused by too thin model."
I have printed files and sliced models with these settings since 2016 in prusaslicer, so it's not that the model is too thin. I have had success with a MK2S and prusaslicer. As you can see in the pictures.

**Comment by SaltWei** [comment]
> but I need the printer to slow down for overhangs because I can't use supports for this print.

I have printed your 3mf and only changed the settings in my screenshot, and printed successfully.

**Comment by SaltWei** [comment]
OK. The original 3MF has been slowing down already

**Comment by Babsobar** [comment]
> > but I need the printer to slow down for overhangs because I can't use supports for this print.
> 
> I have printed your 3mf and only changed the settings in my screenshot, and printed successfully.

Did your print show the same surface quality? I have no clue what that is due to

**Comment by SaltWei** [comment]
Return to the tickets. The hole is just caused by different settings of slicer.
If you use same classic perimeter generator and disable thin wall, prusa also has problem.

**Comment by Babsobar** [comment]
Hello, 
I have fixed the problem. 
Turns out, none of the setttings were the problem for the holes appearing through the print, after a long investigation and trying out a lot of different settings in prusaslicer and BS Lab, I have figured out that the culprit was the mesh exporting algorithm in Fusion360; even though I had the setting on "Refined". The algorithm was making bad faces at the spot were the holes were appearing. While Arachne was able to fix the holes, the resulting prints were not very good in surface quality compared to classic slicing.

I am printing a new print as we speak and will keep you updated on the surface quality.

Thank you for your time and patience.
